### PR TITLE
feat(nixosConfigurations): add dweb-reverse-tls-proxy

### PR DESCRIPTION
**Pull request by steveej**
for testing purposes, this machine is used to reverse-proxy into the machines that will be used at dweb.

~~the zerotier id has been omitted from this commit as it needs to be added as a secret and we currently don't have a way to manage these in this repo.~~

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -65,11 +72,17 @@
         config,
         self',
         inputs',
+        pkgs,
```
Comment (steveej) [inline flake.nix:79]: remove

File: modules/flake-parts/nixosConfigurations.dweb-reverse-tls-proxy/configuration.nix [hunk 2]
```diff
@@ -0,0 +1,344 @@
...
+  config,
+  lib,
+  inputs,
+  self,
+  pkgs,
+  ...
+}: let
+  ipv4 = "5.78.43.185";
+  ipv6Prefix = "2a01:4ff:1f0:872a";
+  fqdn1domain = "holochain.noosphere.life";
```
Comment (steveej) [inline modules/flake-parts/nixosConfigurations.dweb-reverse-tls-proxy/configuration.nix]: remove

File: modules/flake-parts/nixosConfigurations.dweb-reverse-tls-proxy/configuration.nix [hunk 3]
```diff
@@ -0,0 +1,344 @@
...
+      ];
+    };
+  };
+
+  system.stateVersion = "23.05";
+
+  ### ZeroTier
+  services.zerotierone = {
+    enable = lib.mkDefault true;
+    joinNetworks = [
```
Comment (steveej) [inline modules/flake-parts/nixosConfigurations.dweb-reverse-tls-proxy/configuration.nix]: add as a secret

File: modules/flake-parts/nixosConfigurations.dweb-reverse-tls-proxy/configuration.nix [hunk 4]
```diff
@@ -0,0 +1,344 @@
...
+  };
+
+  services.bind = {
+    enable = true;
+    extraConfig = ''
+      include "/var/lib/secrets/*-dnskeys.conf";
+    '';
+    zones = [
+      {
+        name = fqdn1domain;
```
Comment (steveej) [inline modules/flake-parts/nixosConfigurations.dweb-reverse-tls-proxy/configuration.nix]: remove

File: modules/flake-parts/nixosConfigurations.dweb-reverse-tls-proxy/configuration.nix [hunk 5]
```diff
@@ -0,0 +1,344 @@
...
+        extraConfig = "allow-update { key rfc2136key.${fqdn2domain}.; };";
+      }
+    ];
+  };
+
+
+  security.acme = {
+    acceptTerms = true;
+    defaults = {
+      email = "mail@noosphere.life";
```
Comment (steveej) [inline modules/flake-parts/nixosConfigurations.dweb-reverse-tls-proxy/configuration.nix]: change to an @holochain.org email.

@evangineer which one can we use here?

**Review by steveej — COMMENTED** [review]
- [x] remove all references to `noosphere.life`

**Review by ThetaSinner — APPROVED** [review]

**Pull request merged**